internal/call/call.go

```diff
@@ -86,6 +86,15 @@ func (info *Info) AddArgs(ctx context.Context, args ...logf.Marshaler) {
 	info.Args = append(info.Args, args...)
 }
 
+// ApplyOpts applies CallOption functions to the call Info object.
+func (info *Info) ApplyOpts(ctx context.Context, args ...logf.Marshaler) {
+	for _, a := range args {
+		if opt, ok := a.(Option); ok {
```

Comment:
this is such a better interface than wrapping function calls all centered around a context. the only thing i'm a little lost on is why pass args as all types that implement the `logf.Marshaler` interface when what you actually want is only one concrete type - the `Option` interface. Is that just forwards thinking? if so, I think its worth a comment because it's not immediately obvious

Reply:
yeah i probably should've read VKs comment first. that makes sense. comment is a nit, if you think it's unnecessary thats fine w/me

Reply:
Sure, I've added godoc comment